### PR TITLE
tf 1.3 upgrade - bump all module versions in the terraform-cloud chart

### PR DESCRIPTION
**Pull request by easterbrennan**
This bumps all the module versions as part of our work to upgrade to tf 1.3

### CODE REVIEW

File: charts/terraform-cloud/tests/__snapshot__/irsa_workspace_tags_test.yaml.snap [hunk 1]
```diff
@@ -38,7 +38,7 @@ Test IRSA application and component tags default to name:
         - sourceableName: dev-eu-west-1-cluster1-test-namespace-mntl-test-app-s3
       secretsMountPath: /tmp/secrets
       sshKeyID: mintel-ssh
-      terraformVersion: 1.2.9
+      terraformVersion: "1.3"
```
Comment (nabadger) [inline charts/terraform-cloud/tests/__snapshot__/irsa_workspace_tags_test.yaml.snap:41]: Should we be pinning to a patch release here too (1.3.something) ? 

Also wondering why this is quoted this time around...

File: charts/terraform-cloud/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [v1.0.0] - 2024-04-25
```
[the remark this answers is not in the document]
Comment (easterbrennan) [reply charts/terraform-cloud/CHANGELOG.md:8]: yeah I just thought as I've done a major bump for some of the modules this should be a major.

File: charts/terraform-cloud/tests/__snapshot__/irsa_workspace_tags_test.yaml.snap (hunk 1 above)
Comment (easterbrennan) [reply charts/terraform-cloud/tests/__snapshot__/irsa_workspace_tags_test.yaml.snap:41]: yeah not sure what happened with those quotes... I've removed them.  I've also made it 1.3.0 as suggested.

File: charts/terraform-cloud/CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nabadger) [reply charts/terraform-cloud/CHANGELOG.md:8]: ok fair enough

File: charts/terraform-cloud/tests/__snapshot__/irsa_workspace_tags_test.yaml.snap (hunk 1 above)
Comment (nabadger) [reply charts/terraform-cloud/tests/__snapshot__/irsa_workspace_tags_test.yaml.snap]: might be 1.3.10 (will need to check)

Comment (easterbrennan) [reply charts/terraform-cloud/tests/__snapshot__/irsa_workspace_tags_test.yaml.snap]: yeah it's 1.3.10... updated

**Review by nabadger — APPROVED** [review]

**Pull request merged**